client/functions.lua

```diff
@@ -151,45 +181,45 @@ local function lockpickCallback(vehicle, plate, isAdvancedLockedpick, maxDistanc
     end
 end
 
-local lockpickingSemaphore = 0
+local lockpickingSemaphore = 0 -- semaphore
```

Comment:
Is the purpose of this semaphore to prevent calling lockpickDoor more than once? If so, I think a boolean lock would be simpler.

Reply:
if that's enough, no problem. With many processes it would not be enough

Reply:
i changed to islockpickingProcessLocked

Reply:
> if that's enough, no problem. With many processes it would not be enough

What do you mean by this? Shouldn't only one thread be able to hold the lock at a time?

Reply:
In “true” operating system threads, there is a risk that another process will pass a blocking condition before reassigning the state of a variable, so it is necessary to ensure the atomicity of the operation. I don't know enough about lua to know how events will behave when called multiple times. Such a problem was experienced by my colleague at work, in C# today.